client/websocket/websocket.go

```diff
@@ -235,13 +250,13 @@ type marketSyncer struct {
 
 // newMarketSyncer is the constructor for a marketSyncer, returned as a running
 // *dex.StartStopWaiter.
-func newMarketSyncer(ctx context.Context, cl *wsClient, feed *core.BookFeed, log dex.Logger) *dex.StartStopWaiter {
+func newMarketSyncer(cl *wsClient, feed *core.BookFeed, log dex.Logger) *dex.StartStopWaiter {
 	ssWaiter := dex.NewStartStopWaiter(&marketSyncer{
 		feed: feed,
 		cl:   cl,
 		log:  log,
 	})
-	ssWaiter.Start(ctx)
+	ssWaiter.Start(context.Background()) // wrapping Run with a cancel bound to Stop
```

Comment:
The context provided to newMarketSyncer was not needed.  The manager of this syncer (`websocker.(*Server).connect`) uses it as a `StartStopWaiter` via `Stop` and `WaitForShutdown`.